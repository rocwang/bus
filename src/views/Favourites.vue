<template>
  <PanelFavourites
    :isCollapsedInitially="isCollapsedInitially"
    :favouritesWithTrips="favouritesWithTrips"
  />
</template>

<script>
import PanelFavourites from "../components/PanelFavourites";
import { favouritesWithTrips$ } from "../store/trips";
import { actionViewFavourites$ } from "../store/actions";

export default {
  name: "Favourites",
  components: { PanelFavourites },
  props: {
    isCollapsedInitially: {
      type: Boolean,
      default: true
    }
  },
  subscriptions() {
    return {
      favouritesWithTrips: favouritesWithTrips$
    };
  },
  beforeRouteEnter(to, from, next) {
    actionViewFavourites$.next();
    next();
  }
};
</script>
