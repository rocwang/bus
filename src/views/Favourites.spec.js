import { shallowMount } from "@vue/test-utils";
import Favourites from "./Favourites";
import "fake-indexeddb/auto";

describe("Favourites.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Favourites);
  });

  it("matches the snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has the property isCollapsedInitially", () => {
    expect(wrapper.vm.isCollapsedInitially).toBe(true);
  });

  it.skip("triggers the view favourites action before route enter", () => {});
});
