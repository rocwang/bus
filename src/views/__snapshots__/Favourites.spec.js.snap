// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites.vue matches the snapshot 1`] = `
<panelfavourites-stub
  favouriteswithtrips=""
  iscollapsedinitially="true"
/>
`;
